iron-overlay-behavior.html

```diff
@@ -549,23 +570,113 @@
     },
 
     /**
-     * Executes a callback on the next animation frame, overriding any previous
-     * callback awaiting for the next animation frame. e.g.
-     * `__onNextAnimationFrame(callback1) && __onNextAnimationFrame(callback2)`;
-     * `callback1` will never be invoked.
-     * @param {!Function} callback Its `this` parameter is the overlay itself.
+     * Delays the execution of a callback to the next animation frame.
+     * To cancel a previous job, pass callback = undefined
+     * @param {!string} jobname
+     * @param {Function=} callback Always bound to `this`
+     * @private
+     */
+    __deraf: function(jobname, callback) {
+      var rafs = this.__rafs;
+      if (rafs[jobname]) {
+        cancelAnimationFrame(rafs[jobname]);
+      }
+      if (callback) {
+        rafs[jobname] = requestAnimationFrame(function nextAnimationFrame() {
+          rafs[jobname] = null;
+          callback.call(this);
+        }.bind(this));
+      } else {
+        rafs[jobname] = null;
+      }
+    },
+
+    /**
+     * @param {boolean} isAttached
+     * @param {boolean} opened
+     * @param {string=} scrollAction
+     * @private
+     */
+    __updateScrollObservers: function(isAttached, opened, scrollAction) {
+      if (!isAttached || !opened || !this.__isValidScrollAction(scrollAction)) {
+        Polymer.IronScrollManager.removeScrollLock(this);
+        document.removeEventListener('scroll', this.__onCaptureScroll, {
+          passive: true
+        });
+      } else {
+        if (scrollAction === 'lock') {
+          this.__saveScrollPosition();
+          Polymer.IronScrollManager.pushScrollLock(this);
+        }
+        document.addEventListener('scroll', this.__onCaptureScroll, {
+          passive: true
+        });
+      }
+    },
+
+    /**
+     * @param {string=} scrollAction
+     * @return {boolean}
      * @private
      */
-    __onNextAnimationFrame: function(callback) {
-      if (this.__raf) {
-        window.cancelAnimationFrame(this.__raf);
+    __isValidScrollAction: function(scrollAction) {
```

Comment:
I was thinking to have a possible value of `scrollAction` to be `custom` or directly be the name of the callback to be invoked, but I refrained from doing it in this PR for now.
I'd like to see if the 3 actions provided already cover most of the cases and avoid exposing features that haven't been requested.